--- a/gcj/math/BigInteger.java	2011-01-11 19:46:05.000000000 +0000
+++ b/gcj/math/BigInteger.java	2011-07-30 16:35:50.109911630 +0100
@@ -36,7 +36,7 @@
 exception statement from your version. */
 
 
-package java.math;
+package gcj.math;
 
 import gnu.classpath.Configuration;
 
@@ -65,6 +65,7 @@
 public class BigInteger extends Number implements Comparable<BigInteger>
 {
   private static final Logger log = Logger.getLogger(BigInteger.class.getName());
+  public static boolean usingNative() { return USING_NATIVE; }
 
   /** All integers are stored in 2's-complement form.
    * If words == null, the ival is the value of this BigInteger.
